src/xcode/ENA/ENA/Source/Scenes/FileScanner/FileScannerHUD.swift

```diff
@@ -0,0 +1,73 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import UIKit
+
+class FileScannerHUD {
+
+	// MARK: - Init
+
+	init?(
+		execute: @escaping () -> Void
+	) {
+		guard let window = UIApplication.shared.keyWindow else {
+			Log.error("Failed to init - not window found", log: .fileScanner)
+			return nil
+		}
+		self.window = window
+		self.hudView = HUDView()
+		self.execute = execute
+
+		setupView()
+	}
+
+	// MARK: - Internal
+
+	func show() {
+		window.isUserInteractionEnabled = false
+		let animator = UIViewPropertyAnimator(duration: duration, curve: .easeIn) { [weak self] in
+			self?.hudView.alpha = 1.0
+		}
+		animator.addCompletion { [weak self] _ in
+			self?.execute()
+		}
+		animator.startAnimation()
+	}
+
+	func hide() {
+		let animator = UIViewPropertyAnimator(duration: duration, curve: .easeIn) { [weak self] in
+			self?.hudView.alpha = 0.0
+		}
+		animator.addCompletion { [weak self] _ in
+			self?.window.isUserInteractionEnabled = true
+			self?.hudView.removeFromSuperview()
+		}
+		animator.startAnimation()
+	}
+
+	// MARK: - Private
+
+	private let window: UIWindow
+	private let hudView: HUDView
+	private let execute: () -> Void
+	private let duration = 0.45
+
+	private func setupView() {
+		hudView.translatesAutoresizingMaskIntoConstraints = false
+		hudView.isUserInteractionEnabled = false
+		window.addSubview(hudView)
```

Comment:
I guess there is no other way to add the view to the view hierarchy at the view controller level?
To me it looks hacky to add the view to the window directly (also without a view controller).
Just have questions in mind like: 
What about the view controller lifecycle? The `hudView` is not managed under the lifecycle.
What about the user dismissing the view controller in some untracked way. Will the hud be stuck at the window level?
What about some other logic dismissing the view controller, maybe some system logic or some hidden logic in our code (which maybe will be added at a later time)?

Reply:
Please have a look on the changes I made